### ISSUE TITLE
Logging breaks if you make cli requests to two separate sub projects

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#1061]
**Describe the bug**
I was syncing my neutrino node this morning having logs come in just fine to my `$HOME/testnet3/bitcoin-s.log` file, however when i made another rpc call during the sync, (bitcoin-s-cli getnewaddress), it appears that all of the logging that was happening in the chain project related to my IBD broke. I could not get them to restart. 

**To Reproduce**
Watch your `.bitcoin-s/testnet3/bitcoin-s.log` file
Start syncing your node
Make a rpc call to another project (such as `getnewaddress`)
Observe how other projects logs seem to break.

**Expected behavior**
I expected logging to work for all projects even if rpc calls are made that are related to other projects.

This is possibly related to #807

**Comment by nkohen** [comment]
Called `getfilterheadercount`, then `getnewaddress` then `getfilterheadercount` again and got these logs:
```
2020-01-29T21:29:03 DEBUG [chain-verification] Querying for filter header count
2020-01-29T21:29:03 DEBUG [chain-verification] getFilterHeaderCount result: count=1664264
2020-01-29T21:30:45 DEBUG [WALLET] Getting new External adddress for m/84'/1'/0'
2020-01-29T21:30:45 DEBUG [WALLET] Did not find previous address, next=m/84'/1'/0'/0/0
2020-01-29T21:30:45 DEBUG [WALLET] Writing SegWitAddressDb(m/84'/1'/0'/0/0,ECPublicKey(030dcf5200b5a18a36c8f3d5b03050b580175ad0dd37fc9cd5715c6959d8182dab),Sha256Hash160DigestImpl(3acd7e5bf0d994670d06c5b9e83ec2cef0aff3de),Bech32Address(tb1q8txhuklsmx2xwrgxcku7s0kzemc2lu779n45jr),P2WPKHWitnessV0(List(030dcf5200b5a18a36c8f3d5b03050b580175ad0dd37fc9cd5715c6959d8182dab, )),wpkh(Sha256Hash160DigestImpl(3acd7e5bf0d994670d06c5b9e83ec2cef0aff3de))) to DB
2020-01-29T21:30:45 DEBUG [WALLET] Got External address Bech32Address(tb1q8txhuklsmx2xwrgxcku7s0kzemc2lu779n45jr) at key path m/84'/1'/0'/0/0 with pubkey ECPublicKey(030dcf5200b5a18a36c8f3d5b03050b580175ad0dd37fc9cd5715c6959d8182dab)
```
followed by nothing, meaning no log for the second `getfilterheadercount` call which I did get earlier

**Comment by Christewart** [comment]
For more information on current logging configuration in bitcoin-s: 

https://github.com/bitcoin-s/bitcoin-s/blob/master/docs/contributing.md#working-on-bitcoin-s-applications

https://github.com/bitcoin-s/bitcoin-s/blob/master/docs/contributing.md#logging-when-working-on-bitcoin-s-tests

akka logging:

https://github.com/bitcoin-s/bitcoin-s/blob/master/docs/contributing.md#akka-logging

**Comment by Christewart** [comment]
I think a good starting point for this issue is here, this is where we configure the logging: 

https://github.com/bitcoin-s/bitcoin-s/blob/372a19e9ff65f8e17ed98889aab3d427a86c47bc/db-commons/src/main/scala/org/bitcoins/db/AppLoggers.scala#L92

**Comment by nkohen** [comment]
So the problem is that we were creating a new `Logger` for each `LoggerKind` and there can only be one logger per file: causing loggers to shutdown when a new one was created for a different kind.

**Comment by nkohen** [comment]
One solution is to have just one `Logger` with many `FileAppender`s, in which case the `FileAppender`s must be in prudent mode which doesn't allow setting the file field. Additionally I couldn't find how to change the encoder to look at the appender's name, though I am pretty sure this can be done using properties. (see http://logback.qos.ch/manual/layouts.html#conversionWord)

But as this doesn't let you choose the file, I gave up on this approach.

**Comment by nkohen** [comment]
Instead, I attempted to have one `Logger` and only one `FileAppender` that all `LoggerKind`s use, but where before any call to `logger.info` or `logger.error` etc. you first set the MDC (Mapped Diagnostic Context) to have a context field called `name` which the encoder can read.

My attempt at this is implemented on #1086
